File: solutions/ruby/grains/3/grains.rb
module Grains
  MAX_SQUARE = 64
  BASE = 2

  def self.square(num)
    raise ArgumentError, "square must be between 1 and #{MAX_SQUARE}" \
      unless (1..MAX_SQUARE).include?(num)

    BASE**(num - 1)
  end

  def self.total = BASE**MAX_SQUARE - 1
end

File: solutions/python/grains/2/grains.py
MIN_SQUARE = 1
MAX_SQUARE = 64

def square(number):
    if not MIN_SQUARE <= number <= MAX_SQUARE:
        raise ValueError(f"square must be between {MIN_SQUARE} and {MAX_SQUARE}")
    return 1 << (number - 1)  # same as 2**(number - 1)


def total():
    # Sum of grains on all 64 squares
    return (1 << MAX_SQUARE) - 1
